test-assignments/frontend-react-assignment/README.md

```diff
@@ -16,14 +16,16 @@ Please submit the base project as your initial commit and implement the tasks in
 
 - Create a new component which does a basic pagination.
   - This component should contain 2 buttons (Prev, Next) which can paginate the colleague list in the correct direction. The list should only show 5 elements at a time.
-  - Hint: Try to use a Higher Order Component. Do NOT use a 3rd party library for the pagination feature itself but feel free to use any other package for the implementation.
+  - Try to make the pagination as reusable as possible. Do NOT use a 3rd party library for the pagination feature itself but feel free to use any other package for the implementation.
 - Collect the favorite colleagues in a separated list called the “favorites list” already seen on the UI.
   - Hint: The colleague objects already have a `favorite` boolean field.
   - Optional task: Make this list also pageable.
 - Add a new input field on the UI to be able to record the title for a new colleague.
   - This field’s value should be presented in the colleague list.
 - Write unit tests for every new code you write, with a testing framework of your choice.
   - Hint: Our recommended unit testing framework is Jest.
+- Think about the UX of your implementation and make it as user-friendly as possible.
```

Comment:
added this section because it's an important part when we review and I think it's more transparent if we mention this is something we are paying attention to.

Reply:
How about giving the design options instead of asking for it? We implement it based on the design, not by creating it from scratch.

Reply:
While you are right that we follow the design for implementation, some cases are often missing, such as validations, loading state handling, etc. Developers need to understand how their implementation will work, can the users run into edge cases that are not covered by the design, and if yes, then either propose a solution or discuss them with the product.

Reply:
You described an insufficient flow, where "some cases often miss". I got the feeling that you want a random third-party to manage this instead of us. We should work on that, not the candidate.

This is a homework assignment with a limited time to test the candidate's coding skills. I would not ask the candidate to be a developer, designer, product manager, etc.

Reply:
discussed in person, I'll rephrase this sentence to make it a bit more specific